### PR TITLE
Ja/save apis

### PR DESCRIPTION
**Pull request by monkey-jeff**
This PR Lets us build the SDK from the files in this repo.  This changes the workflow slightly

`npm run build` will now always build with the APIs in our local repo

`npm run updateApis` will now download the APIs and add them to the repo

`npm publish` no longer updates the version but instead uses the version from each package.json

There is no longer a env variable for EXCHANGE_DOWNLOAD as that is no longer needed.

Logic that pertains to interactions with exchange have been moved to the exchange-connector package. (Generator no longer downloads the files, exchange-connector does it all by itself)

Because the release process would no longer require github access I also removed a bunch of code from the .circleci configuration that won't be needed.

### CODE REVIEW

**Comment by monkey-jeff** [comment]
NOTE: This is ready to start reviewing.  I would suggest adding a filter to disable the viewing of raml and json files when reviewing.

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,59 +41,61 @@ jobs:
...
       - build-12:
-        filters:
-          branches:
-            ignore:
-              - /^release.*/
-  build-test-and-deploy:
+          filters:
+            tags:
+              ignore: /^v.*/
+  build-test-deploy:
```
Comment (kalagarraj) [inline .circleci/config.yml]: This workflow will get executed for every tagged commit, 

For example one of the checks in this PR https://circleci.com/gh/SalesforceCommerceCloud/commerce-sdk/937?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link actually published the package from an unapproved branch...

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,59 +41,61 @@ jobs:
...
+            branches:
+              ignore: /.*/
+      - build-12:
+          filters:
+            tags:
+              only: /^v.*/
+            branches:
+              ignore: /.*/
+      - confirm-publish-if-tagged:
+          filters:
```
Comment (kalagarraj) [inline .circleci/config.yml]: duplicate filters can be removed for approval step, require build and build12 will take care of filtering

File: .circleci/config.yml [hunk 3]
```diff
@@ -36,59 +41,61 @@ jobs:
...
-              only:
-                - /^release.*/
-      - hold:
+              ignore: /.*/
           type: approval
           requires:
             - build
+            - build-12
       - publish:
+          filters:
```
Comment (kalagarraj) [inline .circleci/config.yml]: filters can be removed, requires:- confirm-publish-if-tagged will take of those..

File: packages/exchange-connector/test/exchangeTools.test.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    {
+      id: "8888888/test-api-4/1.0.0",
+      name: "Test API 4",
+      groupId: "8888888",
+      assetId: "test-api-4",
+      version: "1.0.0"
+    }
+  ];
+
+  it("Group Apis (All same category)", () => {
```
Comment (kalagarraj) [inline packages/exchange-connector/test/exchangeTools.test.ts]: Not clear on what the test is doing...Input is clear, expected result is missing..

Rename tests to 

"<<Expected Result>> on <<given input>>"

File: packages/exchange-connector/src/exchangeTools.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
```
Comment (unandyala) [inline packages/exchange-connector/src/exchangeTools.ts]: Should be 2020 instead of 2019 in the copyright

File: .circleci/config.yml (hunk 1 above)
Comment (monkey-jeff) [reply .circleci/config.yml]: This wasn't unintentional.  This was an intentional test of the deployment path.  It only runs this on tagged commits

File: packages/exchange-connector/src/exchangeTools.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { RestApi } from "./exchangeTypes";
+import _ from "lodash";
+
+/**
+ * @description Group APIs by a category in exchange
+ * @export
+ * @param {RestApi[]} apis
```
Comment (unandyala) [inline packages/exchange-connector/src/exchangeTools.ts]: can you add brief description of each input parameter

File: .circleci/config.yml (hunk 2 above)
Comment (monkey-jeff) [reply .circleci/config.yml]: I had issues with this.  I initially didn't have it but then it wasn't running the step for some reason.  I didn't want to play iwth it much more as this worked and I did find something If you look at https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag there is an example where they have the filter at each step so I just used their example for this.

File: .circleci/config.yml (hunk 3 above)
Comment (monkey-jeff) [reply .circleci/config.yml]: See above comment

File: packages/exchange-connector/test/exchangeTools.test.ts (hunk 4 above)
Comment (monkey-jeff) [reply packages/exchange-connector/test/exchangeTools.test.ts]: Rename tests to?  Can you provide a most specific example please?

File: packages/exchange-connector/gulpfile.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+          } else {
+            api.version = null;
+            return api;
+          }
+        }
+      )
+    );
+  });
+  return Promise.all(promises).then((deployedApis: RestApi[]) => {
+    // createVersionFile(deployedApis, config);
```
Comment (unandyala) [inline packages/exchange-connector/gulpfile.ts]: delete the commented code?

File: .circleci/config.yml (hunk 1 above)
Comment (kalagarraj) [reply .circleci/config.yml]: The problem is unless we publish, PRs containing tagged commits cannot be merged to master.. Is this what we really want?

File: packages/generator/src/renderer.ts [hunk 8]
```diff
@@ -41,7 +40,7 @@ import {
   WebApiBaseUnitWithEncodesModel
 } from "webapi-parser";
 import _ from "lodash";
-import { RestApi } from "@commerce-apps/exchange-connector";
+import { RestApi, groupByCategory } from "@commerce-apps/exchange-connector";
```
Comment (gkhunger-sf) [inline packages/generator/src/renderer.ts]: groupByCategory is unused

File: packages/exchange-connector/test/exchangeTools.test.ts (hunk 4 above)
Comment (kalagarraj) [reply packages/exchange-connector/test/exchangeTools.test.ts]: weird, I had the following in the comment,

Rename tests to 

"Expected Result on given input".

File: .circleci/config.yml (hunk 2 above)
Comment (kalagarraj) [reply .circleci/config.yml]: Ok, Removing the filters from subsequent steps should work, 

not a big deal anyways. Just the config will be more cleaner..

File: .circleci/config.yml (hunk 1 above)
Comment (monkey-jeff) [reply .circleci/config.yml]: Why can't they be merged to master?

File: packages/exchange-connector/src/exchangeDirectoryParser.ts [hunk 9]
```diff
@@ -21,7 +22,10 @@ function getFiles(directory): fs.Dirent[] {
  * zip files are usually downloaded from Anypoint exchange
  * @param directory
  */
-export function extractFiles(directory: string): Promise<void> {
+export function extractFiles(
+  directory: string,
+  removeFiles = true
```
[the remark this answers is not in the document]
Comment (monkey-jeff) [reply packages/exchange-connector/src/exchangeDirectoryParser.ts]: This was an old tsdoc description I hadn't written.  Updating it to be more inline with our other docs now.

File: .circleci/config.yml (hunk 1 above)
Comment (kalagarraj) [reply .circleci/config.yml]: As discussed in our standup, we are not doing annotated commits. we can merge to master.. This comment can be resolved

**Pull request merged**